### PR TITLE
WD-14648 - refactor: dedupe the matching and filtering functions

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Replace the identical filtering and matching functions with generic methods inside `PanelTableForm`.

## Details

https://warthogs.atlassian.net/browse/WD-14648

### CODE REVIEW

File: src/components/EntityTable/EntityTable.tsx [hunk 1]
```diff
@@ -10,6 +10,21 @@ import EntityTablePagination from "./EntityTablePagination";
 import type { TableEntity } from "./types";
 import { Label, type Props } from "./types";
 
+const getProps = (id: string) => {
```
Comment (huwshimi) [inline src/components/EntityTable/EntityTable.tsx:13]: Drive-by dedupe this function as well.

**Review by vladimir-cucu — APPROVED** [review]
